File: test_cafekiosk/src/main/java/com/testcafekiosk/spring/domain/product/Product.java
//package com.testcafekiosk.spring.domain.product;
//
//import com.testcafekiosk.spring.domain.BaseEntity;
//import jakarta.persistence.*;
//import lombok.AccessLevel;
//import lombok.Builder;
//import lombok.Getter;
//import lombok.NoArgsConstructor;
//
//@Entity
//@Getter
//@NoArgsConstructor(access = AccessLevel.PROTECTED)
//public class Product extends BaseEntity {
//
//    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
//    private Long id;
//
//    private String productNumber;
//
//    @Enumerated(EnumType.STRING)
//    private ProductType type;
//
//    @Enumerated(EnumType.STRING)
//    private ProductSellingType sellingStatus;
//
//    private String name;
//
//    private int price;
//
//    @Builder
//    private Product(String productNumber, ProductType type, ProductSellingType sellingStatus, String name, int price) {
//        this.productNumber = productNumber;
//        this.type = type;
//        this.sellingStatus = sellingStatus;
//        this.name = name;
//        this.price = price;
//    }
//}

File: test_cafekiosk/src/main/kotlin/com/testcafekiosk/spring/domain/product/Product.kt
package com.testcafekiosk.spring.domain.product

import com.testcafekiosk.spring.domain.BaseEntity
import jakarta.persistence.*

@Entity
class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val productNumber: String,

    @Enumerated(EnumType.STRING)
    val type: ProductType,

    @Enumerated(EnumType.STRING)
    val sellingStatus: ProductSellingStatus,

    val name: String,

    val price: Int,

    ) : BaseEntity() {
}